check_run/check_run/report/ach_prenote/ach_prenote.js

```diff
@@ -0,0 +1,188 @@
+			getEditor: this.get_editing_object.bind(this),
+		})
+	},
+	get_editing_object(colIndex, rowIndex, value, parent) {
+		const control = this.render_editing_input(colIndex, value, parent)
+		if (!control) return false
+		control.df.change = event => {
+			frappe.query_report.page.set_indicator('Unsaved', 'orange')
+			changedItems[rowIndex] = event?.currentTarget.value || null
+			this.add_save_button(frappe.query_report)
+			control.set_focus()
+		}
+		try {
+			return {
+				initValue: async value => {
+					return control.set_value(value)
+				},
+				setValue: value => {
+					if (!value) {
+						return control.get_value()
+					} else {
+						return control.set_value(value)
+					}
+				},
+				getValue: async () => {
+					return control.get_value()
+				},
+			}
+		} catch (error) {
+			console.log(error)
+		}
+	},
+	render_editing_input(colIndex, value, parent) {
+		const col = frappe.query_report.datatable.getColumn(colIndex)
+		let control = null
+		control = frappe.ui.form.make_control({
+			df: col,
+			parent: parent,
+			render_input: true,
+		})
+		control.set_value(value || '')
+		control.toggle_label(false)
+		control.toggle_description(false)
+		return control
+	},
+	formatter: (value, row, column, data, default_formatter) => {
+		value = default_formatter(value, row, column, data)
+		if (data && column.fieldname == 'party_name') {
+			value = `<a href="/app/${frappe.scrub(data.party_type, '-')}/${data.party}">${data.party_name}</a>`
+		}
+		return value
+	},
+	add_save_button: reportview => {
+		if (reportview.saveButton) {
+			return
+		}
+		reportview.saveButton = frappe.query_report.page.add_button(
+			__('Save Changes'),
+			() => {
+				let data = []
+				let row
+				for (const [rowIndex, updated_date] of Object.entries(changedItems)) {
+					row = frappe.query_report.data[rowIndex]
+					row.account_details_validated = updated_date
+					data.push(row)
+				}
+				frappe
+					.xcall('check_run.check_run.report.ach_prenote.ach_prenote.update_validated_dates', { data: data })
+					.then(r => {
+						reportview.saveButton.remove()
+						reportview.saveButton = null
+						changedItems = {}
+						reportview.data = []
+						frappe.query_report.page.set_indicator('', '')
+						// this.get_datatable_options(frappe.query_report.datatable.options)
+						reportview.refresh()
+						frappe.query_report.page.set_indicator('', '')
+						frappe.show_alert(__('Updated Validated Dates'), 5)
+					})
+			},
+			{ btn_class: 'btn-primary' }
+		)
+	},
+}
+
+function generate_ach_prenote() {
+	return new Promise(resolve => {
+		let dialog = new frappe.ui.Dialog({
+			title: __('Please provide additional details'),
+			fields: [
+				{
+					label: 'Check Run Settings',
+					fieldname: 'check_run_settings',
+					fieldtype: 'Link',
+					options: 'Check Run Settings',
+					reqd: 1,
+				},
+				{
+					fieldtype: 'Currency',
+					label: __('ACH Amount'),
+					fieldname: 'ach_amount',
+					default: 0.05,
```

Comment:
Minor point - we mention the amount is usually 50 cents in the docs, then use 5 cents here - consider conforming the two?

Reply:
That one is a hallucination - I have no idea what a normal value is